internal/pkg/fieldtags/analyzer.go

```diff
@@ -42,17 +39,13 @@ var patterns sourcePatterns = []keyValue{
 	{"levee", "source"},
 }
 
-// TaggedFields is named type representing a slice of TaggedFields.
-type TaggedFields []TaggedField
+// TaggedFields is a map from types.Object to bool.
+// It can be used to determine whether a field is a tagged Source field.
+type TaggedFields map[types.Object]bool
```

Comment:
I maybe commented on the last one, but I don't see why we need to export both `TaggedFields` and `ResultType`.  I have no opinion about which it should be.

Reply:
I find having a `ResultType` symbol is useful when you want to get the analyzer's result, e.g. `pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)`.

Apart from that I agree that it's not super useful, and may contribute to noise. Also it duplicates documentation, and in this case it also led to the documentation for `ResultType` being out of date.

I've removed `TaggedFields`. It's not super useful to have two aliases, and I think `ResultType` is the more useful of the two.

Since this "pattern" was also in use in the fieldpropagators analyzer, I've made a PR for that one too, for consistency: #78